2226/2/versus.rb

```diff
@@ -0,0 +1,192 @@
+require 'terminal-table'
+require 'fuzzy_match'
+require 'optparse'
+require 'russian'
+require 'yaml'
+
+# This class is used for storing data
+class DataStorage
+  def self.show_all_data
+    Dir.glob('rap-battles/*').each_with_object({}) do |file_name, hash|
+      hash[file_name] = File.read(file_name)
+    end
+  end
+
+  def self.list_all_rapers
+    filter_rapers.flatten.uniq
+  end
+
+  def self.filter_rapers_from_the_list(file_name)
+    filter = file_name.sub(/ротив_/, '').sub(/_\(.*/, '')
+    filter.sub(%r/^rap-battles\/_/, '').sub(/_$/, '')
+  end
+
+  def self.filter_rapers
+    show_all_data.inject([]) do |array, (file_name, _file_content)|
+      filtered_data = filter_rapers_from_the_list(file_name).split('_п')
+      array << filtered_data
+    end
+  end
+end
+
+# This class is used for single raper methods
+class Raper
+  SWEAR_WORDS = YAML.load_file('config.yml')['SWEAR_WORDS_ARRAY'].join('|')
+
+  def self.avg_number_of_battles(name)
+    DataStorage.show_all_data.inject(0) do |counter, (file_name, _file_content)|
+      counter + file_name.scan(name).size
+    end
+  end
+
+  def self.number_of_battles(name)
+    avg_number_of_battles(name) / 2
+  end
+
+  def self.number_of_swear_words(name)
+    real_name = name
+    DataStorage.show_all_data.inject(0) do |counter, (file_name, file_content)|
+      counter += file_content.scan(/#{SWEAR_WORDS}/).size if file_name.match(real_name)
+      counter
+    end
+  end
+
+  def self.average_number_swearing_words_in_battle(name)
+    average = number_of_swear_words(name).fdiv(number_of_battles(name))
+    average.round(2)
```

Comment:
do not mix data processing and presentation. This rounding should be somewhere in table drawing method.